### PR TITLE
[cpp_info] [refact] Rewrite text generator parser

### PR DESCRIPTION
**Pull request by jgsogo**
Changelog: omit
Docs: omit

Parser reconstructs the objects used to dump that information (new test to check that content is not modified after reading and writing again): we read `xxxx_paths`, make the paths relative and assign that to `xxxxdirs` attributes.

### CODE REVIEW

**Review by memsharded — COMMENTED** [review]
Sure it is worth the effort? IMHO this generator should completely disappear in Conan 2.0

**Comment by jgsogo** [comment]
> Sure it is worth the effort? IMHO this generator should completely disappear in Conan 2.0

It's already (almost) done, and it is needed in order to change existing CppInfo objects for the new ones.

**Review by danimtb — APPROVED** [review]

File: conans/test/functional/command/test_package_test.py [hunk 1]
```diff
@@ -129,7 +129,7 @@ def check_version_test(self):
             from conans import ConanFile
             class Dep(ConanFile):
                 def package_info(self):
-                    self.cpp_info.name = "MyDep"
+                    self.cpp_info.names["txt"] = "MyDep"
```
Comment (memsharded) [inline conans/test/functional/command/test_package_test.py]: Is it really necessary to change this test? Or could a new test be added instead?

**Review by memsharded — APPROVED** [review]
This has a non-zero risk of 🍔  

But I like the clean-ups, so go ahead, I am hungry.... 😂

File: conans/test/functional/command/test_package_test.py (hunk 1 above)
Comment (jgsogo) [reply conans/test/functional/command/test_package_test.py]: no, I can revert this change 👍

**Comment by jgsogo** [comment]
I really trust on the _idempotent_ test I've added (`test_read_write`), it's my safety network 🎪

**Pull request merged**